lighthouse-core/config/v2/config.js

```diff
@@ -0,0 +1,186 @@
+const path = require('path');
+const log = require('../../lib/log');
+
+const defaultConfigPath = './default.json';
+const defaultConfigJson = require(defaultConfigPath);
+
+const _flatten = arr => [].concat(...arr);
+const _subtract = (setA, setB) => Array.from(setA).filter(x => !setB.has(x));
+
+class ConfigV2 {
+  constructor(configJson, configPath) {
+    if (!configJson) {
+      configJson = defaultConfigJson;
+      configPath = path.resolve(__dirname, defaultConfigPath);
+    }
+
+    // Perform a shallow clone so we can adjust gatherers and audits
+    configJson = Object.assign({}, configJson);
+
+    // Resolve the paths of audits and gatherers
+    configJson.gatherers = ConfigV2.resolvePaths(configJson.gatherers, configPath, [
+      path.join(__dirname, '../../gather/gatherers'),
+    ]);
+    configJson.audits = ConfigV2.resolvePaths(configJson.audits, configPath, [
+      path.join(__dirname, '../../audits'),
+    ]);
+
+    // Extend only after our paths have been resolved
+    configJson = ConfigV2.extendIfNecessary(configJson, configPath);
```

Comment:
> it looks like you're just matching current behavior here?

The intent behind several of the config changes was specifically to enable clear and simple extension not needing logic or fancy behavior of any sort, so this is the final behavior as far as I'm concerned. That's what the structure of the new config enables :)